index.js

```diff
@@ -40,12 +40,47 @@ function isValidSchema (schema, name) {
   }
 }
 
-function mergeLocation (location, key) {
-  return {
-    schema: location.schema[key],
-    schemaId: location.schemaId,
-    jsonPointer: location.jsonPointer + '/' + key,
-    isValidated: location.isValidated
+class Location {
```

Comment:
Did you evaluate the `URL` class?

Reply:
Sorry, I didn't get you.

Reply:
The [URL](https://nodejs.org/api/url.html#class-url) class has many convenient methods to calculate if the URL is absolute and relative and compose it.

Looking at this `Location` class, I have afraid that we are going to replicate the `URL` class in future.

So I wanted to point it out to reduce the burden you could face by developing it

Reply:
I don't think they are the same and `URL` is really slow.
The usage here are only `string` concatenation.